online-dashboard-api/app/Http/Controllers/AuthenticationController.php

```diff
@@ -0,0 +1,205 @@
+            }
+
+            //generate access token using helper function
+            $tokenData = generateAccessToken($user, $request->password);
+
+            //checking if token generation failed
+            if (isset($tokenData['error'])) {
+                return response()->json([
+                    'message' => 'Token generation failed',
+                    'error' => $tokenData['error'],
+                    'error_description' => $tokenData['error_description'] ?? ' ',
+                ], 400);
+            }
+
+            // Login the user
+            Auth::login($user);
+
+            // Start a session manually
+            if (! Session::isStarted()) {
+                Session::start();
+            }
+
+            //success response if tokens are generated successfully
+            return response()->json([
+                'message' => 'Sccessufully logged in',
+                'token_type' => $tokenData['token_type'],
+                'access_token' => $tokenData['access_token'],
+                'refresh_token' => $tokenData['refresh_token'],
+                'expires_in' => $tokenData['expires_in'],
+                'user' => Auth::user(),
+            ], 200);
+
+        } catch (Throwable $e) {
+            return response()->json([
+                'message' => 'An error occurred during login or token generation',
+                'error' => $e->getMessage(),
+            ], 400);
+        }
+
+    }
+
+    // Revoke the access token
+    public function logout(Request $request)
+    {
+        //destroying the session
+        Session::invalidate();
+
+        //regenerate the session ID to prevent session fixation attacks
+        Session::regenerateToken();
+
+        info('inside the logout');
+        $user = Auth::user();
+        info($user);
+        $tokens = Auth::user()->tokens;
+        // Check if user exists and has an active token
+        if ($user && $tokens) {
+
+            foreach ($tokens as $token) {
+                $token->revoke();
+                $refreshTokenRepository = app(RefreshTokenRepository::class);
+                $refreshTokenRepository->revokeRefreshTokensByAccessTokenId($token->id);
+            }
+
+            //logout the user
+            // Auth::logout();
+
+            return response()->json(['message' => 'Successfully logged out'], 200);
+        } else {
+            return response()->json(['message' => 'No active access token found for the user'], 400);
+        }
+
+    }
+
+    public function refreshAccessToken(RefreshRequest $request)
+    {
+
+        $refreshToken = $request->refresh_token;
+
+        try {
+            // Create an instance of AccessTokenController to handle the request
+            $accessTokenController = app(AccessTokenController::class);
+
+            // Create a Psr17Factory instance to generate PSR-7 compatible request
+            $psr17Factory = new Psr17Factory;
```

Comment:
Move this entire server request to a seperate file and get the URL http://127.0.0.1:8000/oauth/token from the config